addons/crm/models/crm_lead.py

```diff
@@ -745,6 +771,18 @@ def create(self, vals_list):
                 vals['website'] = self.env['res.partner']._clean_website(vals['website'])
         leads = super().create(vals_list)
 
+        commercial_partner_ids = [vals['commercial_partner_id'] for vals in vals_list if vals.get('commercial_partner_id')]
+        CommercialPartners = self.env['res.partner'].with_prefetch(commercial_partner_ids)
+        for lead, lead_vals in zip(leads, vals_list):
+            if not lead_vals.get('partner_id') and lead_vals.get('commercial_partner_id'):
+                if 'default_partner_id' in self.env.context:
+                    continue
+                commercial_partner = CommercialPartners.browse(lead_vals['commercial_partner_id'])
+                if (lead.phone or lead.email_from) and (lead.phone != commercial_partner.phone or lead.email_from != commercial_partner.email):
```

Comment:
A bit like done above: we should probably try to compare normalized values (email_normalized and phone_sanitized). Indeed formatting may differ.  Both partner and lead models inherit from mail.thread.phone (crm here, and partner in phone_validation which is a dependency of this module) so comparison can be done on stored fields directly.